pkg/controller/deployment/credentials_request.go

```
@@ -55,71 +57,101 @@ func withCloudCredentials(secretsInformer coreinformersv1.SecretInformer, infraI
+		},
+	}
+	volumeMount := &corev1.VolumeMount{
+		Name:      cloudCredentialsVolumeName,
+		MountPath: gcpCredentialsDir,
+	}
+	return volume, volumeMount, nil
+}
+
+func applyCloudCredentialsToDeployment(deployment *appsv1.Deployment, volume *corev1.Volume, volumeMount *corev1.VolumeMount, envVar *corev1.EnvVar) error {
+	if len(deployment.Spec.Template.Spec.Containers) == 0 {
+		return fmt.Errorf("deployment %s/%s has no containers, cannot apply cloud credentials", deployment.GetNamespace(), deployment.GetName())
```

golangci-lint (2.5.0)

[error] do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"deployment %s/%s has no containers, cannot apply cloud credentials\", deployment.GetNamespace(), deployment.GetName())"

(err113)